demo_nodes_cpp/src/topics/talker_raw.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rcutils/cmdline_parser.h"
+#include "rcutils/snprintf.h"
+
+#include "std_msgs/msg/string.hpp"
+
+using namespace std::chrono_literals;
+
+void print_usage()
+{
+  printf("Usage for talker app:\n");
+  printf("talker [-t topic_name] [-h]\n");
+  printf("options:\n");
+  printf("-h : Print this help function.\n");
+  printf("-t topic_name : Specify the topic on which to publish. Defaults to chatter.\n");
+}
+
+class RawTalker : public rclcpp::Node
+{
+public:
+  explicit RawTalker(const std::string & topic_name)
+  : Node("raw_talker")
+  {
+    raw_msg_.buffer_length = 24;
+    raw_msg_.buffer = new char[raw_msg_.buffer_length];
```

Comment:
While the `delete` in the destructor is sufficient for the demo code to be "correct" I would suggest to use a smart pointer. Otherwise this might encourage people to copy-n-paste this code which is "incomplete" without a copy constructor and assignment operator.

Reply:
agreed, a smart pointer would be nice, but I don't think it's more user friendly. I would rather say it's even more error prone to make copy-paste mistakes as you can't simply instantiate a shared ptr with that c-struct. You have to pass a custom deleter function / callable object with the smart pointer to clean up correctly. Does this make sense?

Reply:
A `std::unique_ptr` is specialized for arrays and should be fine to use (as of C++11) without a custom deleter afaik (see http://en.cppreference.com/w/cpp/memory/unique_ptr).